src/Highway.Data.EntityFramework/Contexts/ReadonlyDataContext.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Data.Entity;
+using System.Linq;
+
+using Common.Logging;
+using Common.Logging.Simple;
+
+using Highway.Data.EntityFramework;
+
+namespace Highway.Data
+{
+    public class ReadonlyDataContext : IReadonlyEntityDataContext
+    {
+        private readonly ReadonlyDbContext _context;
+
```

Comment:
We now take create a ReadonlyDbContext for composition, rather than inheriting from it, this hiding the .Database property.  ReadonlyDbContext is internal.